src/peripheral/nvic.rs

```diff
@@ -118,12 +153,40 @@ impl RegisterBlock {
     ///
     /// NOTE See `get_priority` method for an explanation of how NVIC priorities
     /// work.
+    ///
+    /// On ARMv6-M, updating an interrupt priority requires a read-modify-write
```

Comment:
I think this is not an issue with scoped singletons, at least if we change the signature to `&mut self`. With scoped singletons if the singleton (peripheral) in owned by an execution context then that context has `&mut -` access to the peripheral and this is not an issue. If the peripheral is shared between two, or more, execution contexts then some mechanism must be in place to (safely and temporarily) yield a `&mut-` reference to the register block to one of the contexts -- again, no issue there.

I think it's still a good idea to mention that the operation is an atomic write on ARMv7 and a RMW operation on ARMv6.